example_queries/get_service_name.rs.ref

```diff
@@ -181,83 +237,76 @@ impl Filter {
-            }
-        }
-        let prop_to_trace = properties.join(",");
-        trace = utils::generate_trace_graph_from_headers(path, prop_to_trace);
-        return trace;
-    }
-
     pub fn on_incoming_requests(&mut self, mut x: Rpc) -> Vec<Rpc> {
-        let mut prop_str;
-        prop_str = format!("{whoami}.{property}=={value}",
-                                                      whoami=&self.whoami.as_ref().unwrap(),
-                                                      property="service_name",
-                                                      value=self.filter_state["node.metadata.WORKLOAD_NAME"].string_data.as_ref().unwrap().to_string());
-                                             
-        if x.headers.contains_key("properties_service_name") {
-            if !x.headers["properties_service_name"].contains(&prop_str) { // don't add our properties if they have already been added
-                x.headers.get_mut(&"properties_service_name".to_string()).unwrap().push_str(",");
-                x.headers.get_mut(&"properties_service_name".to_string()).unwrap().push_str(&prop_str);
-            }
-        }
-        else {
-            x.headers.insert("properties_service_name".to_string(), prop_str);
-        }
-         prop_str = format!("{whoami}.{property}=={value}",
-                                                      whoami=&self.whoami.as_ref().unwrap(),
-                                                      property="service_name",
-                                                      value=self.filter_state["node.metadata.WORKLOAD_NAME"].string_data.as_ref().unwrap().to_string());
-                                             
-        if x.headers.contains_key("properties_service_name") {
-            if !x.headers["properties_service_name"].contains(&prop_str) { // don't add our properties if they have already been added
-                x.headers.get_mut(&"properties_service_name".to_string()).unwrap().push_str(",");
-                x.headers.get_mut(&"properties_service_name".to_string()).unwrap().push_str(&prop_str);
+        // Fetch ferried data
+        let mut ferried_data: FerriedData;
+        if !x.headers.contains_key("ferried_data") {
+            ferried_data = FerriedData::default();
+        } else {
+            match serde_json::from_str(&x.headers["ferried_data"]) {
+                Ok(fd) => { ferried_data = fd; }
+                Err(e) => {
+                    print!("ERROR:  could not translate stored data to json string: {0}\n", e);
+                    return vec![x];
+                }
             }
         }
-        else {
-            x.headers.insert("properties_service_name".to_string(), prop_str);
-        }
-         
-        self.store_headers(x.clone());
-        return vec!(x);
+
+        // Insert properties to collect
+        let mut prop_tuple;
+        prop_tuple = (self.whoami.as_ref().unwrap().to_string(),
```

Comment:
This is unreadable and should be simplified. Maybe a struct? That is basically a named tuple.

Reply:
Or just use variables.

Reply:
That makes sense.  It is now a struct.